### PR TITLE
Fix bug in wasm table size calculation

### PR DESCRIPTION
**Pull request by dschuff**
When mapping split/homogenous tables to the single wasm table, the size
counting code didn't handle empty tables. These only occur with `-s
RELOCATABLE=1`, which results in an additional 'X' signature (which has
all the functions) and empty tables for the other signatures.

This resulted in `Module['wasmTableSize']` (which feeds into the size of
the `WebAssembly.Table` object) not matching the initial size of the table
import declaration in the wasm module. Currently V8 has a
limitation that it throws a LinkError in this case, while SM is happy
with it. In any case, since this mismatch is not intentional, it should
be fixed.

### CODE REVIEW

File: emscripten.py [hunk 1]
```diff
@@ -709,10 +709,13 @@ def keyfunc(other):
       asm_setup += 'var setTempRet0 = Runtime.setTempRet0, getTempRet0 = Runtime.getTempRet0;\n'
 
     if settings['BINARYEN']:
-      table_size = sum(map(lambda table: table.count(',') + 1, last_forwarded_json['Functions']['tables'].values()))
-      asm_setup += "\nModule['wasmTableSize'] = %d;\n" % table_size
+      def table_size(table):
+        s = table.count(',')
+        return s + 1 if s > 0 else 0
```
Comment (kripken) [inline emscripten.py]: this doesn't seem right. If the table is `[just_this_one_function]` then we should return 1. I.e. `s` is set to 0 if the table size is either 0 or 1.

to detect an empty table, we should probably check for `table == '[]'`. I don't think you'll need to erase whitespace, we don't emit any

Comment (dschuff) [inline emscripten.py]: er.. right. so `table` is actually the whole `var FUNCTION_TABLE_i = [...]` expression, so... yay parsing!

File: emscripten.py [hunk 2]
```diff
@@ -709,10 +709,17 @@ def keyfunc(other):
       asm_setup += 'var setTempRet0 = Runtime.setTempRet0, getTempRet0 = Runtime.getTempRet0;\n'
 
     if settings['BINARYEN']:
-      table_size = sum(map(lambda table: table.count(',') + 1, last_forwarded_json['Functions']['tables'].values()))
-      asm_setup += "\nModule['wasmTableSize'] = %d;\n" % table_size
+      def table_size(table):
+        t = table[table.index('[') + 1: table.index(']')].split(',')
```
Comment (dschuff) [inline emscripten.py]: Ok.

**Comment by kripken** [comment]
lgtm, thanks. I also did some local testing of this and saw no problems.

**Pull request merged**